hooks/scheduler.js

```diff
@@ -63,24 +65,27 @@ class Scheduler {
     return this.done;
   }
 
-  /** Polls for hooks that need to be scheduled and handles them in a loop */
   async poll() {
-    while(!this.stopping) {
-      // Get all hooks that have a scheduled date that is earlier than now
-      var hooks = await this.Hook.scan({
-        nextScheduledDate:  base.Entity.Op.lessThan(new Date())
-      }, {});
+    // Get all hooks that have a scheduled date that is earlier than now
+    var hooks = await this.Hook.scan({
+      nextScheduledDate:  base.Entity.op.lessThan(new Date())
+    }, {});
```

Comment:
This needs to handle paging... azure may page if crossing partition boundaries, or being slow...
It's easy to do this with, handler: https://github.com/taskcluster/azure-entities/blob/c2a386a5a4d33d791ce8ef27c7a03ba4962d868b/src/entity.js#L1156-L1159

Example:
```js
var hooks = await this.Hook.scan({
  nextScheduledDate:  base.Entity.op.lessThan(new Date())
}, {
  limit: 100, // lets be reasonable about not making too many requests in parallel
  handler: (hook) => this.handleHook(hook)
});
```
Again, I would set `nextScheduledDate` to `new Date(MAX_INT)` if this hook isn't to be scheduled... Or something like that...
Actually, stackoverflow says `new Date(8640000000000000)` we can probably make that constant somewhere :)
see: http://stackoverflow.com/a/11526569